src/server/saved_objects/client/lib/create_id_query.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Finds a document by either its v5 or v6 format
+ *
+ * @param type The documents type
+ * @param id The documents id or legacy id
+**/
+export function createIdQuery({ type, id }) {
+  return {
+    version: true,
+    size: 1,
+    query: {
+      bool: {
+        should: [
+          // v5 document
```

Comment:
I guess this explains my earlier id-comment.

Isn't there a possible conflict here? 

E.g., suppose I create a new dashboard after re-indexing with the same human readable id I had from an older v5 dashboard.

```
curl -X POST -H "Content-Type: application/json" -H "kbn-xsrf: 6.0.0-alpha3" -d '{ "version": "6.0.0-alpha3", "objects": [{ "id": "test", "type": "dashboard", "version": 1, "attributes": { "title": "My Test Dashboard From V6" }}]}' http://localhost:5601/api/kibana/dashboards/import
{"objects":[{"id":"test","type":"doc","version":1,"attributes":{"title":"My Test Dashboard From V6"}}]}
```

This creates a new dashboard:

```
curl http://localhost:5601/apdashboardbjects/ 
{"saved_objects":[{"id":"AV0UtNI1urO7MVHxeLol","type":"dashboard","version":1,"attributes":{"title":"My Test Dashboard"}},{"id":"test","type":"dashboard","version":1,"attributes":{"title":"My Test Dashboard From V6"}}],"total":2,"per_page":20,"page":1}
```

But when I now query by id, I get the old v5 dashboard like earlier.

```
curl http://localhost:5601/api/saved_objects/dashboard/test
{"id":"AV0UtNI1urO7MVHxeLol","type":"dashboard","version":1,"attributes":{"title":"My Test Dashboard"}}
```